ios/brave-ios/Sources/Data/models/PlaylistItem.swift

```diff
@@ -160,6 +174,41 @@ final public class PlaylistItem: NSManagedObject, CRUD, Identifiable {
     }
   }
 
+  public static func addItems(
+    _ items: [PlaylistInfo],
+    folderUUID: String? = nil,
+    cachedData: Data?,
+    completion: (() -> Void)? = nil
+  ) {
+    DataController.perform(context: .new(inMemory: false), save: false) { context in
+      for item in items {
+        let playlistItem = PlaylistItem(
+          context: context,
+          name: item.name,
+          pageTitle: item.pageTitle,
+          pageSrc: item.pageSrc,
+          cachedData: cachedData ?? Data(),
+          duration: item.duration,
+          mimeType: item.mimeType,
+          mediaSrc: item.src
+        )
+        playlistItem.order = item.order
+        playlistItem.uuid = item.tagId
+        playlistItem.playlistFolder = PlaylistFolder.getFolder(
```

Comment:
You can retrieve this once outside the for-loop and assign it here I think?

Reply:
And maybe create  another method for initializing the playlist item and use it for both addItem and a addItems methods

Reply:
Honestly this method is just for unit tests anyways 😅